EcommerceStore.Application/Services/ProductCategoryService.cs

```diff
@@ -0,0 +1,131 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using EcommerceStore.Application.Exceptions;
+using EcommerceStore.Application.Interfaces;
+using EcommerceStore.Application.Models.InputModels;
+using EcommerceStore.Application.Models.ViewModels;
+using EcommerceStore.Domain.Entities;
+using EcommerceStore.Domain.Interfaces;
+
+namespace EcommerceStore.Application.Services
+{
+    public class ProductCategoryService : IProductCategoryService
+    {
+        private readonly IProductCategoryRepository _productCategoryRepository;
+
+        public ProductCategoryService(IProductCategoryRepository productCategoryRepository)
+        {
+            _productCategoryRepository = productCategoryRepository;
+        }
+
+        public async Task CreateProductCategoryAsync(ProductCategoryInputModel productCategoryInputModel)
+        {
+            var existingProductCategory = await _productCategoryRepository.GetByNameAsync(productCategoryInputModel.Name);
+
+            if (existingProductCategory != null)
+                throw new ValidationException(AlreadyExistExceptionMessages.ProductCategoryAlreadyExists);
+
+            var productCategory = new ProductCategory
+            {
+                Name = productCategoryInputModel.Name,
+                ParentCategoryId = productCategoryInputModel.ParentCategoryId
+            };
+
+            productCategory.ProductCategorySections.Add(new ProductCategorySection
```

Comment:
let's move link category -> section(s) to separate method in sections service  ( add validation that if parent category exists and belongs to sections A and B, child category can be also linked only to sections A and B, not to C and D)
also add method in this service for linking subcategory to existing category